### PR TITLE
Event image URLS showing localhost in production environment

### LINKED ISSUE #5307: Urls like original-image-url get changed after creating events using API
**Describe the bug**
Urls like `originial-image-url`are getting changed in the response shown after using POST request to create them. API used `api.eventyay.com/v1/events`. Below is the event to be posted raw data and corresponding response.

Event to be posted:-
```
{
  "data": {
    "attributes": {
      "name": "Music & Voice Therapy with Michele Chong",
      "external-event-url": "http://example.com",
      "starts-at": "2018-10-12T23:59:59.123456+00:00",
      "ends-at": "2018-10-20T23:59:59.123456+00:00",
      "timezone": "UTC",
      "latitude": "1.29456789",
      "longitude": "103.85",
      "logo-url": "https://upload.wikimedia.org/wikipedia/commons/8/88/Music-segno.png",
      "location-name": "Raffles Place, Singapore",
      "searchable-location-name": "Singapore",
      "description": "An oft cited notable example of the dilemma in defining music is a work entitled 4'33'' (1952) by the American composer John Cage (1912–1992). The written score has three movements and directs the performer(s) to appear on stage, indicate by gesture or other means when the piece begins, then make no sound and only mark sections and the end by gesture. What is heard are only whatever ambient sounds may occur in the room. Some argue this is not music because, for example, it contains no sounds that are conventionally considered musical",
      "original-image-url": "https://steemitimages.com/DQmWG96SKvDEx98pB4T34CB7UqgkZDRawHNaDsoETzLmQ57/1240e483-c8b9-4d0d-8b48-c8c245f2c05e.jpg",
      "organizer-name": "Musical Organizers",
      "is-map-shown": "true",
      "organizer-description": "Good Organizer!",
      "is-sessions-speakers-enabled": "true",
      "privacy": "public",
      "state": "draft",
      "average-rating": null,
      "ticket-url": "http://example.com",
      "code-of-conduct": "example",
      "is-ticketing-enabled": "true",
      "payment-country": "US",
      "payment-currency": "USD",
      "paypal-email": "example@example.com",
      "is-tax-enabled": "true",
      "can-pay-by-paypal": "true",
      "can-pay-by-stripe": "true",
      "can-pay-by-cheque": "true",
      "can-pay-by-bank": "true",
      "can-pay-onsite": "true",
      "cheque-details": "example",
      "bank-details": "example",
      "onsite-details": "example",
      "is-sponsors-enabled": "false",
      "has-organizer-info": "false",
      "order-expiry-time": "10"
    },
    "type": "event"
  }
}
```

Response:-

```
{
    "data": {
        "type": "event",
        "attributes": {
            "can-pay-by-stripe": true,
            "organizer-description": "Good Organizer!",
            "schedule-published-on": null,
            "cheque-details": "example",
            "state": "draft",
            "thumbnail-image-url": "http://localhost:5000/static/media/events/11/thumbnail/aGdud2NMcT/c1449a43-ca6b-42c3-ada7-e357ac6b21b5.jpg",
            "location-name": "Raffles Place, Singapore",
            "organizer-name": "Musical Organizers",
            "can-pay-by-bank": true,
            "starts-at": "2018-10-12T23:59:59.123456+00:00",
            "timezone": "UTC",
            "is-sponsors-enabled": false,
            "can-pay-by-paypal": true,
            "large-image-url": "http://localhost:5000/static/media/events/11/large/QXN6enM3Nj/f3ddb7c7-f036-443f-aca6-feb18a968a03.jpg",
            "has-organizer-info": false,
            "is-map-shown": true,
            "original-image-url": "http://localhost:5000/static/media/events/11/original/cFlISFBQb2/2eeef3bc-5b24-4b25-a094-616ced922432.jpg",
            "ticket-url": "http://example.com",
            "icon-image-url": "http://localhost:5000/static/media/events/11/icon/cFhCcnBiRV/fc2d8186-c9ff-4fc7-a723-87a21c34808d.jpg",
            "xcal-url": null,
            "onsite-details": "example",
            "average-rating": null,
            "identifier": "b5f87a50",
            "latitude": 1.29456789,
            "is-sessions-speakers-enabled": true,
            "pentabarf-url": null,
            "code-of-conduct": "example",
            "is-ticketing-enabled": true,
            "bank-details": "example",
            "longitude": 103.85,
            "external-event-url": "http://example.com",
            "ical-url": null,
            "searchable-location-name": "Singapore",
            "can-pay-by-cheque": true,
            "paypal-email": "example@example.com",
            "ends-at": "2018-10-20T23:59:59.123456+00:00",
            "privacy": "public",
            "payment-currency": "USD",
            "name": "Music & Voice Therapy with Michele Chong",
            "deleted-at": null,
            "logo-url": "https://upload.wikimedia.org/wikipedia/commons/8/88/Music-segno.png",
            "is-tax-enabled": true,
            "payment-country": "US",
            "can-pay-onsite": true,
            "description": "An oft cited notable example of the dilemma in defining music is a work entitled 4'33'' (1952) by the American composer John Cage (1912–1992). The written score has three movements and directs the performer(s) to appear on stage, indicate by gesture or other means when the piece begins, then make no sound and only mark sections and the end by gesture. What is heard are only whatever ambient sounds may occur in the room. Some argue this is not music because, for example, it contains no sounds that are conventionally considered musical",
            "created-at": "2018-08-07T15:27:14.583517+00:00"
        },
        "relationships": {
            "tickets": {
                "links": {
                    "self": "/v1/events/11/relationships/tickets",
                    "related": "/v1/events/11/tickets"
                }
            },
            "sessions": {
                "links": {
                    "self": "/v1/events/11/relationships/sessions",
                    "related": "/v1/events/11/sessions"
                }
            },
            "speakers": {
                "links": {
                    "self": "/v1/events/11/relationships/speakers",
                    "related": "/v1/events/11/speakers"
                }
            },
            "moderators": {
                "links": {
                    "self": "/v1/events/11/relationships/moderators",
                    "related": "/v1/users"
                }
            },
            "registrars": {
                "links": {
                    "self": "/v1/events/11/relationships/registrars",
                    "related": "/v1/users"
                }
            },
            "event-topic": {
                "links": {
                    "self": "/v1/events/11/relationships/event-topic",
                    "related": "/v1/events/11/event-topic"
                }
            },
            "stripe-authorization": {
                "links": {
                    "self": "/v1/stripe-authorizations/11/relationships/event",
                    "related": "/v1/events/11/stripe-authorization"
                }
            },
            "role-invites": {
                "links": {
                    "self": "/v1/events/11/relationships/role-invites",
                    "related": "/v1/events/11/role-invites"
                }
            },
            "orders": {
                "links": {
                    "self": "/v1/events/11/relationships/orders",
                    "related": "/v1/events/11/orders"
                }
            },
            "sponsors": {
                "links": {
                    "self": "/v1/events/11/relationships/sponsors",
                    "related": "/v1/events/11/sponsors"
                }
            },
            "discount-codes": {
                "links": {
                    "self": "/v1/events/11/relationships/discount-codes",
                    "related": "/v1/events/11/discount-codes"
                }
            },
            "event-sub-topic": {
                "links": {
                    "self": "/v1/events/11/relationships/event-sub-topic",
                    "related": "/v1/events/11/event-sub-topic"
                }
            },
            "social-links": {
                "links": {
                    "self": "/v1/events/11/relationships/social-links",
                    "related": "/v1/events/11/social-links"
                }
            },
            "event-type": {
                "links": {
                    "self": "/v1/events/11/relationships/event-type",
                    "related": "/v1/events/11/event-type"
                }
            },
            "event-invoices": {
                "links": {
                    "self": "/v1/events/11/relationships/event-invoices",
                    "related": "/v1/events/11/event-invoices"
                }
            },
            "access-codes": {
                "links": {
                    "self": "/v1/events/11/relationships/access-codes",
                    "related": "/v1/events/11/access-codes"
                }
            },
            "faq-types": {
                "links": {
                    "self": "/v1/events/11/relationships/faq-types",
                    "related": "/v1/events/11/faq-types"
                }
            },
            "speakers-call": {
                "links": {
                    "self": "/v1/events/11/relationships/speakers-call",
                    "related": "/v1/events/11/speakers-call"
                }
            },
            "faqs": {
                "links": {
                    "self": "/v1/events/11/relationships/faqs",
                    "related": "/v1/events/11/faqs"
                }
            },
            "tax": {
                "links": {
                    "self": "/v1/events/11/relationships/tax",
                    "related": "/v1/events/11/tax"
                }
            },
            "event-copyright": {
                "links": {
                    "self": "/v1/events/11/relationships/event-copyright",
                    "related": "/v1/events/11/event-copyright"
                }
            },
            "custom-forms": {
                "links": {
                    "self": "/v1/events/11/relationships/custom-forms",
                    "related": "/v1/events/11/custom-forms"
                }
            },
            "attendees": {
                "links": {
                    "self": "/v1/events/11/relationships/attendees",
                    "related": "/v1/events/11/attendees"
                }
            },
            "tracks": {
                "links": {
                    "self": "/v1/events/11/relationships/tracks",
                    "related": "/v1/events/11/tracks"
                }
            },
            "ticket-tags": {
                "links": {
                    "self": "/v1/events/11/relationships/ticket-tags",
                    "related": "/v1/events/11/ticket-tags"
                }
            },
            "coorganizers": {
                "links": {
                    "self": "/v1/events/11/relationships/coorganizers",
                    "related": "/v1/users"
                }
            },
            "feedbacks": {
                "links": {
                    "self": "/v1/events/11/relationships/feedbacks",
                    "related": "/v1/events/11/feedbacks"
                }
            },
            "track-organizers": {
                "links": {
                    "self": "/v1/events/11/relationships/track-organizers",
                    "related": "/v1/users"
                }
            },
            "microlocations": {
                "links": {
                    "self": "/v1/events/11/relationships/microlocations",
                    "related": "/v1/events/11/microlocations"
                }
            },
            "organizers": {
                "links": {
                    "self": "/v1/events/11/relationships/organizers",
                    "related": "/v1/users"
                }
            },
            "session-types": {
                "links": {
                    "self": "/v1/events/11/relationships/session-types",
                    "related": "/v1/events/11/session-types"
                }
            }
        },
        "id": "11",
        "links": {
            "self": "/v1/events/11"
        }
    },
    "links": {
        "self": "/v1/events/11"
    },
    "jsonapi": {
        "version": "1.0"
    }
}
```

### PR DESCRIPTION
**Pull request by bhaveshAn**
Fixes #5307 Event image URLS showing localhost in production environment

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:

Fixes #5307 Event image URLS showing localhost in production environment

#### Changes proposed in this pull request:

- Fixes #5307 Event image URLS showing localhost in production environment

screenshot from 2018-08-17 20-27-25
screenshot from 2018-08-17 20-28-10

### CODE REVIEW

File: app/api/helpers/storage.py [hunk 1]
```diff
@@ -179,7 +183,12 @@ def upload_local(uploaded_file, key, **kwargs):
 
 def create_url(request_url, file_relative_path):
     """Generates the URL of an uploaded file."""
-    url = urlparse(request_url)
+
+    if env('APP_CONFIG') == 'config.ProductionConfig' or \
+            os.getenv('APP_CONFIG') == 'config.ProductionConfig':
+        url = urlparse('http://api.eventyay.com')
```
Comment (iamareebjamal) [inline app/api/helpers/storage.py]: As suspected, your previous methods(all) were wrong ways to produce the URLs

This too is wrong. What about the server hosted on heroku? Or self hosted ones. This is called "HARDCODING" which is the next bad thing about a server after deliberately throwing an exception

Comment (bhaveshAn) [reply app/api/helpers/storage.py]: I find two solutions of this, either using HARDCODING way or SERVER_NAME env.
You are not favour of using SERVER_NAME env too.
Please suggest a way which can fully solve the issue without these two ways.

> What about the server hosted on heroku? 

Heroku using DevelopmentConfig

screenshot from 2018-08-10 20-33-11

**Comment by iamareebjamal** [comment]
I wrote detailed method on how to achieve this in the original PR, but you just merged it anyway before anyone approving it.

**Comment by bhaveshAn** [comment]
@iamareebjamal If talking about to use request.url then its already being used here https://github.com/fossasia/open-event-server/blob/development/app/api/helpers/storage.py#LC177 but not giving us api.eventyay.com when creating image urls from production server.

**Comment by iamareebjamal** [comment]
The `SERVER_NAME` is not working correctly as it prevents local setup. Try it and see.

**Comment by bhaveshAn** [comment]
@iamareebjamal Yes, SERVER_NAME doesn't work. I have changed implementation. Please check it once.

**Comment by iamareebjamal** [comment]
I don't think there is a need for it in the DB. Can just be there in settings

**Comment by bhaveshAn** [comment]
@iamareebjamal On Travis earlier I was failing the three tests of storage named `test_arbitrary_url` `test_http_url` `test_https_url` so added check of os.getenv(APP_CONFIG) == 'config.TestingConfig'
Have a look.

**Comment by poush** [comment]
I'm not aware with the previous PR so not sure but why don't you pick the server_url from ".env" which is basically from environment vars? 

People can put two vars like

SERVER_URL
TEST_SERVER_URL

and when the environment is testing one, the ".env" parser can append "TEST_" to all keys to get respective value from env vars/file.

**Comment by bhaveshAn** [comment]
> I'm not aware with the previous PR so not sure but why don't you pick the server_url from ".env" which is basically from environment vars?
People can put two vars like
SERVER_URL
TEST_SERVER_URL
and when the environment is testing one, the ".env" parser can append "TEST_" to all keys to get respective value from env vars/file.

@iamareebjamal Can I do this ??

**Comment by bhaveshAn** [comment]
@iamareebjamal @poush Please review.

**Comment by bhaveshAn** [comment]
screenshot from 2018-08-17 20-27-25
screenshot from 2018-08-17 20-28-10

File: app/api/helpers/storage.py [hunk 2]
```diff
@@ -180,10 +182,16 @@ def upload_local(uploaded_file, key, **kwargs):
...
     """Generates the URL of an uploaded file."""
     url = urlparse(request_url)
-
+    if not is_test:
+        if app.config['SERVER_URL']:
+            url = urlparse(app.config['SERVER_URL'])
+        else:
+            pass
+    else:
+        pass
```
Comment (poush) [inline app/api/helpers/storage.py]: Seems good, just one thing, is there no need to load test_Server_url  in this case?

Comment (bhaveshAn) [reply app/api/helpers/storage.py]: I hasn't able to find any reason to use that, for now.
Can you clear it if its really of use.

**Comment by bhaveshAn** [comment]
@Kreijstal Will this implementation be used to get SERVER_URL from .env at api.eventyay.com or just os.getenv is the only solution to this. I am asking this because I wasn't able to find any clone of server on Hetzner which is using .env to get envs.  Hence making change in config.py or env doesn't change anything until we use envs exported in os ??

**Comment by CosmicCoder96** [comment]
@bhaveshAn Update for any changes and get it merged please.

**Comment by bhaveshAn** [comment]
Already requested reviews

File: app/api/helpers/storage.py [hunk 3]
```diff
@@ -180,10 +182,16 @@ def upload_local(uploaded_file, key, **kwargs):
     return create_url(request.url, file_relative_path)
 
 
-def create_url(request_url, file_relative_path):
+def create_url(request_url, file_relative_path, is_test=False):
     """Generates the URL of an uploaded file."""
     url = urlparse(request_url)
-
+    if not is_test:
```
Comment (poush) [inline app/api/helpers/storage.py]: Yes, this is redundant can be done in simple one `if` condition.

File: app/api/helpers/storage.py (hunk 2 above)
Comment (poush) [reply app/api/helpers/storage.py]: There's no need to write these `pass` statements here.

**Pull request closed without merging**